### PR TITLE
Use more modern metrics reporting with data_stats.

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
Modernize data_stats a bit, so that it prints nicer things. Small refactor. Also speed it up.

**Logs**
```
$ parlai data_stats -t dailydialog
ParlAI - Dialogue Research Platform
22:53:18 | creating task(s): dailydialog
22:53:24 | loaded 22236 episodes with a total of 87170 examples
           avg_utterance_length  tokens  unique_tokens  unique_utterances  utterances
   both                   13.75 2396388          20674              71389      174340
   input                  12.98 1131176          19741              63158       87170
   labels                 14.51 1265212          20673              71388       87170
```

### CODE REVIEW

File: parlai/scripts/data_stats.py [hunk 1]
```diff
@@ -130,54 +119,52 @@ def keep_token(t):
...
+            logging.info(text)
 
     try:
         # print dataset size if available
         logging.info(
             f'loaded {world.num_episodes()} episodes with a total '
             f'of {world.num_examples()} examples'
         )
-    except Exception:
+    except AttributeError:
```
Comment (emilydinan) [inline parlai/scripts/data_stats.py:156]: when does a world not have these attributes?

File: parlai/scripts/data_stats.py [hunk 2]
```diff
@@ -130,54 +119,52 @@ def keep_token(t):
...
         pass
-    return report(world, counts, log_time)
+
+    retval = _report(world, counts)
+    retval.pop('exs')
```
Comment (emilydinan) [inline parlai/scripts/data_stats.py:160]: why do we remove this

Comment (stephenroller) [reply parlai/scripts/data_stats.py:160]: it looked ugly lol

File: parlai/scripts/data_stats.py (hunk 1 above)
Comment (stephenroller) [reply parlai/scripts/data_stats.py:156]: some worlds don't! streaming ones.

**Comment by stephenroller** [comment]
@emilydinan will you stamp?

**Comment by stephenroller** [comment]
This just needs to be reviewed

**Comment by dexterju** [comment]
It looks great, Thanks for adding the tests as well!

**Review by dexterju — APPROVED** [review]

**Pull request merged**